spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe "Customers Show" do 
+  describe "as a visitor" do 
+    before(:each) do
+      @supermarket_1 = Supermarket.create!(name: "Sprouts", location: "CO")
+
+      @item_1 = @supermarket_1.items.create!(name: "Beef", price: 5)
+      @item_2 = @supermarket_1.items.create!(name: "Cheese", price: 3)
+      @item_3 = @supermarket_1.items.create!(name: "Soup", price: 2)
+
+      @customer_1 = Customer.create!(name: "Aaron")
+      @customer_2 = Customer.create!(name: "Chase")
+
+      CustomerItems.create!(customer: @customer_1, item: @item_1)
+      CustomerItems.create!(customer: @customer_1, item: @item_2)
+      CustomerItems.create!(customer: @customer_2, item: @item_2)
+      CustomerItems.create!(customer: @customer_2, item: @item_3)
+    end
+
+    # User Story 1
+    it "displays customer's name and list of customer's items with attributes" do
+      visit "/customers/#{@customer_1.id}"
+
+      expect(page).to have_content("Name: #{@customer_1.name}")
+
+      within(".items-list") do
```

Comment:
Nice use of within blocks here! 

Another option: putting a html/css class on a div inside the iteration you do for the @customer.items.each

  <% @customer.items.each do |item| %>
    <div class = <% "item-#{item.id}" %>>
        <p>Item: <%= item.name %></p>
        <p>Cost: <%= item.price %></p>
        <p>Store: <%= item.supermarket.name %></p></br>
   </div>
 <% end %>
```

Now you have a div with a specific item ID so you can do `within(".item-#{item1.id}")` and get even more granular with your assertions!

Reply:
We should also see expect statements here for item 2's attributes which are already associated to customer 1